litellm/proxy/proxy_server.py

```
@@ -1558,7 +1558,12 @@ async def load_config(  # noqa: PLR0915
         environment_variables = config.get("environment_variables", None)
         if environment_variables:
             for key, value in environment_variables.items():
-                os.environ[key] = str(get_secret(secret_name=key, default_value=value))
+                secret = get_secret(secret_name=key, default_value=value)
+                if isinstance(secret, str):
+                    os.environ[key] = secret
+                else:
+                    from litellm._logging import verbose_logger
```

what happens in the else condition? 

does it just not load in the env var?

---

I don't know what is the best thing to do in this situation, silently converting anything to `str` is quite dangerous as shown by the bug induced here.

Is there any legitimate situation (that is, not caused by a bug in `get_secret` function such as the one fixed in this pull request) where `get_secret` returns something that is not already a `str` and you really want to convert it into a `str`?

---

Hey @user202729 was the bug that None was a 'str' ?

---

The actual bug fix is in the other file `litellm/secret_managers/main.py` — previously, when `secret` is None then the `get_secret` function returns `None`. With this change, when `secret` is None then the `get_secret` function returns `default_value` instead.

Previously the `None` returned gets converted to the string `"None"` which is clearly undesirable. So I double-check it here (`litellm/proxy/proxy_server.py`) just in case some other code path also return something unexpected.

Although the change here is unnecessary to fix the bug.

---

Actually… the new logic still feel wrong. If there's some existing environment variable and the user override it in the config file then the value in the config file should be honored right?

---

okay it was first added in 9943c6d6079974b1f773674236d4ce86d2fd3d9e , let's see